### ISSUE TITLE
Option to add a Nonce to request

### ISSUE THREAD

**Issue by muyiwaoyeniyi** [alexandrtovmach/react-microsoft-login#65]
Would it be possible to add a nonce to the request to mitigate timing attacks with the tokens returned? Also, an option to pass a State prop would be great too.

Thanks.

**Comment by alexandrtovmach** [comment]
Yeah, sounds reasonable. I'll take a look, but any PR or help are welcome
